lib/jsonapi/authorization/default_pundit_authorizer.rb

```diff
@@ -190,6 +190,36 @@ def remove_to_many_relationship(source_record, related_record)
       def remove_to_one_relationship(source_record, related_record)
         raise NotImplementedError
       end
+
+      # Any request including <tt>?include=other-resources</tt>
+      #
+      # This will be called for each has_many relationship if the include goes
+      # deeper than one level until some authorization fails or the include
+      # directive has been travelled completely.
+      #
+      # We can't pass all the records of a +has_many+ association here due to
+      # performance reasons, so the class is passed instead.
+      #
+      # ==== Parameters
+      #
+      # * +record_class+ - The underlying record class for the relationships
+      #                    resource.
+      def include_has_many_resource(record_class)
```

Comment:
Pundi::Scope is applied already in the resource level, so we don't have to do it here. Passing the source relation record here would merely allow custom logic for the authorization check, e.g. it would allow a case such as

```ruby
if source_relation_record.is_a?(Article)
  ::Pundit.authorize(user, record_class, 'include_with_article?')
else
  ::Pundit.authorize(user, record_class, 'index?')
end
```

Right now this authorization method does not have context of what was the source of the include query — that is, if we got here through `?include=author.comments` (`record_class` would be `Comment` in that case), this method does not know that we're including comments for an author record.